From a29509f696227073ecf6a6c74a7b5c45bc21880c Mon Sep 17 00:00:00 2001
From: XScreenSaver owners <xscreensaver-owner@fedoraproject.org>
Date: Wed, 15 Feb 2023 17:25:16 +0900
Subject: [PATCH] switch_page_cb: backport debian fix for DPMS settings issue

Backport: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1031076
Fix the issue that dpms enabled_p option is not preserved
correctly.
---
 driver/demo-Gtk.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/driver/demo-Gtk.c b/driver/demo-Gtk.c
index a84b420..baacf31 100644
--- a/driver/demo-Gtk.c
+++ b/driver/demo-Gtk.c
@@ -1677,6 +1677,7 @@ switch_page_cb (GtkNotebook *notebook, GtkWidget *page,
   state *s = &win->state;
 
   if (s->debug_p) fprintf (stderr, "%s: tab changed\n", blurb());
+  populate_prefs_page (s);
   pref_changed_cb (GTK_WIDGET (notebook), user_data);
 
   /* If we're switching to page 0, schedule the current hack to be run.
-- 
2.39.2

